client/src/app.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var app = angular.module('sunfishApp', ['ngRoute']);
+'$compileProvider',
+    app.config(['$routeProvider', '$compileProvider', function($routeProvider, $compileProvider) {
+        $routeProvider
+        .when('/', {
+            templateUrl: 'views/main/main.html',
+        })
+        .when('/browse/', {
+            controller: "BrowseCtrl",
+            templateUrl: 'views/browse/browse.html',
+        })
+        .when('/search/', {
+            controller: "SearchCtrl",
+            templateUrl: 'views/search/search.html',
+        })
+        .when('/upload/', {
+            controller: "UploadCtrl",
+            templateUrl: 'views/upload/upload.html',
+        })
+        .when('/siafile/:siafileId', {
+            controller: "SiafileCtrl",
+            templateUrl: 'views/siafile/siafile.html',
+        })
+        .otherwise({redirectTo: '/'});
```

Comment:
Is this `.when` detection the reason each html snippet needs to be in its own folder? It seems simpler to have a folder called "markup" with them all in it. Easier browsing.